torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -1061,6 +1108,78 @@ HeuristicSummary::HeuristicSummary(
   recording_ = false;
 }
 
+void HeuristicSummary::validate() const {
+  switch (heuristic_) {
+    case ScheduleHeuristic::PointWise:
+      TORCH_INTERNAL_ASSERT(
+          entry_type_map_.count(EntryType::VECTORIZABLE_INPUTS_AND_OUTPUTS));
+      break;
+    case ScheduleHeuristic::Reduction:
+      TORCH_INTERNAL_ASSERT(entry_type_map_.count(EntryType::REDUCTION_TVS));
+      TORCH_INTERNAL_ASSERT(
+          entry_type_map_.count(EntryType::VECTORIZABLE_INPUTS_AND_OUTPUTS));
+      break;
+    case ScheduleHeuristic::Normalization:
+      TORCH_INTERNAL_ASSERT(entry_type_map_.count(EntryType::REDUCTION_TVS));
+      TORCH_INTERNAL_ASSERT(
+          entry_type_map_.count(EntryType::VECTORIZABLE_INPUTS_AND_OUTPUTS));
+      TORCH_INTERNAL_ASSERT(
+          entry_type_map_.count(EntryType::PERSISTENT_BUFFER_INFO));
+      // If check persistent factor only when persistent buffers needed.
+      auto persistent_buffer_info =
+          entry_type_map_.at(EntryType::PERSISTENT_BUFFER_INFO)
+              ->as<
+                  CompileTimeInfo<HeuristicCompileTime::PersistentBufferInfo>>()
+              ->get();
+      TORCH_INTERNAL_ASSERT(
+          persistent_buffer_info->buffers.empty() ||
+          entry_type_map_.count(EntryType::SCOPE_PERSISTENT_FACTOR_INFO));
+      TORCH_INTERNAL_ASSERT(
+          entry_type_map_.count(EntryType::REDUCTION_TOPOLOGY_INFO));
+      break;
+  }
+}
+
+void HeuristicSummary::insert(HeuristicSummary::EntryOwningPtr new_entry) {
+  TORCH_INTERNAL_ASSERT(
+      recording_, "should only insert entries at recording phase");
+  // Just override when insertion duplicates, equality not checked.
+  entry_type_map_[new_entry->type()] = new_entry.get();
```

Comment:
Should you assert there isn't already an entry here?

Reply:
Currently not asserting because `canSchedule` and `getHeuristics` often write to the same entry, and they should be the same result as they are from the same function called on the same fusion. We could later update this interface to be either more strict or more flexible on overwriting data depending on future use cases.